UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Storage/StorageService.cs

```diff
@@ -0,0 +1,30 @@
+﻿
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using UKHO.ExternalNotificationService.Common.Configuration;
+
+namespace UKHO.ExternalNotificationService.Common.Storage
+{
+    public class StorageService : IStorageService
+    {
+        private readonly IOptions<SubscriptionStorageConfiguration> _storageConfig;
+        public StorageService(IOptions<SubscriptionStorageConfiguration> storageConfig)
+        {
+            _storageConfig = storageConfig;
+        }
+        public string GetStorageAccountConnectionString(string storageAccountName = null, string storageAccountKey = null)
+        {
+            string storageAccountAccessKeyValue = !string.IsNullOrEmpty(storageAccountKey) ? storageAccountKey : _storageConfig.Value.StorageAccountKey;
+            string storageAccountNameValue = !string.IsNullOrEmpty(storageAccountName) ? storageAccountName : _storageConfig.Value.StorageAccountName;
+
+            if (string.IsNullOrWhiteSpace(storageAccountAccessKeyValue))
+            {
+                throw new KeyNotFoundException($"Storage account accesskey not found");
```

Comment:
Thanks for adding the unit test class. One more test needed to cover this path please.

Reply:
Added KeyNotFoundException unit test cases

Reply:
Unit Test cases added for KeyNotFoundException